src/vmm/src/vstate/memory.rs

```diff
@@ -306,28 +306,48 @@ impl GuestMemoryExtension for GuestMemoryMmap {
         let mut writer_offset = 0;
         let page_size = get_page_size().map_err(MemoryError::PageSize)?;
 
+        // Merging KVM dirty pages bitmap into internal firecracker bitmap. Needed to keep track of
+        // KVM dirty pages in case we need to use them again if current dump operation fails.
         self.iter()
             .enumerate()
             .try_for_each(|(slot, region)| {
-                let kvm_bitmap = dirty_bitmap.get(&slot).unwrap();
                 let firecracker_bitmap = region.bitmap();
-                let mut write_size = 0;
-                let mut dirty_batch_start: u64 = 0;
-
+                let kvm_bitmap = dirty_bitmap.get(&slot).unwrap();
                 for (i, v) in kvm_bitmap.iter().enumerate() {
+                    // FIXME: looks like this loop can be replaced with some
```

Comment:
you mean `fc_dirty = fc_dirty | kvm_dirty`? Yeah, I think that would be simpler than iterating over each bit

Reply:
Yes, something like that. Not sure yet what are the types of the `fc_dirty` and `kvm_dirty`.